### ISSUE TITLE
Problems with intersect function

### ISSUE THREAD

**Issue by nidaca** [orientechnologies/orientdb#4605]
I have a simple test case:
- schema:
  a vertex class 'Tag' (with associated cluster #14) and an edge class 'tagged' (from the Tag class to another class).
- queries:
  select intersect(in('tagged')) from [#14:2, #14:78] ... I get an empty 'intersect' field although the two vertices have some common adjacent nodes. If I change a little bit the query I get the correct result:
  select intersect(in('tagged').@rid) from [#14:2, #14:78].

When tested with only one vertex I get the following result-sets (orient studio raw view):

```
select intersect(in('tagged').@rid) from [#14:2]  >>>>>>>
{
    "result": [
        {
            "@type": "d",
            "@rid": "#-2:0",
            "@version": 0,
            "intersect": [
                "#19:51965",
                "#19:51966",
                ............
            ],
            "@fieldTypes": "intersect=e"
        }
    ],
    "notification": "Query executed in 0.273 sec. Returned 1 record(s)"
}  >>>>>>> ok!

select intersect(in('tagged')) from [#14:2]  >>>>>>>
{
    "result": [
        {
            "@type": "d",
            "@rid": "#-2:0",
            "@version": 0,
            "intersect": [
                [
                    "#19:51965",
                    "#19:51966",
                    ............
                ]
            ],
            "@fieldTypes": "intersect=e"
        }
    ],
    "notification": "Query executed in 0.273 sec. Returned 1 record(s)"
}  >>>>>>> seems wrong!
```

Can you guys please check this simple issue / bug ?

**Comment by nidaca** [comment]
... actually I think this is a real bug, not a question :)

**Comment by luigidellaquila** [comment]
Hi @nidaca 
the two results you posted seem to be identical, I guess it's just a cut'n'paste problem...
Anyway I got the point, I'll check it asap

**Comment by nidaca** [comment]
Hi Luigi, what seems like a copy-paste problem (in the second result) is actually what I get from the studio raw view and seems to be the root of the issue. An 'intersect' result collection which contains another one is not what we expect from this query ...

**Comment by luigidellaquila** [comment]
Hi @nidaca 
could you try this and let me know if it produces the expected result?

```sql
select  expand(intersect(in('tagged'))) from [#14:2]
```

**Comment by nidaca** [comment]
yes, tested in studio with no results. but I tested also:
select expand(intersect(in('tagged').@rid)) from #14:2   or   select expand(intersect(in('tagged').@this)) from #14:2, again with no results ... strange ...
if I replace 'intersect' with something else, let's say 'list' I get the correct behavior for all queries:
select expand(list(in('tagged'))) from #14:2

**Comment by luigidellaquila** [comment]
Ok, it's definitely a bug.
Last question: which OrientDB version are you using?

**Comment by nidaca** [comment]
2.0.12 is the version we use

**Comment by ozgursucu** [comment]
Hi @luigidellaquila,
This problem arises orientdb 3.0 again.

**Comment by luigidellaquila** [comment]
Hi @ozgursucu @nidaca 

I just pushed a fix on 2.2.x and develop, could you please check if it fixes the problem?

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi

this issue should be fixed, I'm closing it. Please reopen if you still see a wrong behavior

Thanks

Luigi
